plugins/meta/bandwidth/README.md

```diff
@@ -0,0 +1,112 @@
+# bandwidth plugin
+
+## Overview
+
+This plugin provides a way to use and configure Linux's Traffic control (tc) subystem. tc encompasses the sets of mechanisms and operations by which packets are queued for transmission/reception on a network interface.
+
+This plugin configures a token bucket filter (tbf) queuing discipline (qdisc) on both ingress and egress traffic. Resulting in traffic being shaped when reading / writing.
+
+Due to limitations on tc shaping rules for ingress, this plugin creates an Intermediate Functional Block device (ifb) to redirect packets from the host interface. tc tbf is then applied to the ifb device. The packets that were redirected to the ifb devices, are written OUT (and shaped) to the host interface.
+
+This plugin is only useful when used in addition to other plugins.
+
+## Chaining
+
+The bandwidth plugin applies traffic shaping to interfaces (as described above) created by previously applied plugins.
+
+The following is an example [json configuration list](https://github.com/containernetworking/cni/blob/master/SPEC.md#network-configuration-list-runtime-examples) for creating a `ptp` between the host -> container via veth interfaces, whereby traffic is shaped by the `bandwidth` plugin:
+
+```json
+{
+  "cniVersion": "0.3.1",
+  "name": "mynet",
+  "plugins": [
+    {
+      "name": "my-net",
+      "type": "ptp",
+      "ipMasq": true,
+      "mtu": 512,
+      "ipam": {
+          "type": "host-local",
+          "subnet": "10.0.0.0/24"
+      },
+      "dns": {
+        "nameservers": [ "10.1.0.1" ]
+      }
+    },
+    {
+      "name": "slowdown",
+      "type": "bandwidth",
+      "rate": 123,
+      "burst": 456
+    }
+  ]
+}
+```
+
+
+1. The following is the json payload the container runtime would need to perform during the ADD action:
```

Comment:
You probably don't need to explain plugin chaining in this README.

Reply:
removed plugin chaining examples. pushed.